Update OpenTelemetry Collector depenency to v0.63.1

tpaschalis opened the pull request:
Signed-off-by: Paschalis Tsilias <paschalis.tsilias@grafana.com>

<!--

CONTRIBUTORS GUIDE: https://github.com/grafana/agent/blob/main/docs/developer/contributing.md#updating-the-changelog

If this is your first PR or you have not contributed in a while, we recommend
taking the time to review the guide. It gives helpful instructions for
contributors around things like how to update the changelog.

-->

#### PR Description
This PR updates our dependency to the OpenTelemetry Collector to v0.63.1.

I'm leaving it as draft so that it doesn't get merged before the v0.29.0 RC is cut. Will also add a CHANGELOG entry once the new changelog section has been added for the v0.29 release.

#### Which issue(s) this PR fixes
Fixes #2449

#### Notes to the Reviewer
The code changes are due to the following PRs

* https://github.com/open-telemetry/opentelemetry-collector/commit/e4cfde47582df7bc20106fc0713d0a19b889f2a3 Rename MetricAggregationTemporality to AggregationTemporality
* https://github.com/open-telemetry/opentelemetry-collector/commit/3b70df911f822da13d9fa3478efea7f1d9f03b34 Deprecate pdata Clone methods in favor of CopyTo
* https://github.com/open-telemetry/opentelemetry-collector/commit/2829ed3c4f087065dfaea88778ca87caba5464ad Rename pcommon.Map.PutString to PutStr
* https://github.com/open-telemetry/opentelemetry-collector/commit/9a4333959bbbbb8722b67a92713a0c37bf3a1e13 Split config unmarshaler per config section
* https://github.com/open-telemetry/opentelemetry-collector/commit/3f3ecd2e371040774f0b5adef7f93e83660bf9ff Rename enum zero value constants to have Empty suffix
* https://github.com/open-telemetry/opentelemetry-collector/commit/6e554f2d823b488d5964a335d9bc59995ee9ce64 Rename pmetric.MetricDataPointFlags to pmetric.DataPointFlags

Also, now the collector/exporter/otlpexporter and collector/exporter/otlphttpexporter packages got their own go.mod file and are now treated as separate modules, so two extra replace directives were required for us to continue using our fork.
```
	go.opentelemetry.io/collector/exporter/otlpexporter => github.com/grafana/opentelemetry-collector/exporter/otlpexporter v0.0.0-20221101113646-2be4b341c150
	go.opentelemetry.io/collector/exporter/otlphttpexporter => github.com/grafana/opentelemetry-collector/exporter/otlphttpexporter v0.0.0-20221101113646-2be4b341c150
```

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added (N/A)
- [ ] Tests updated (N/A)

rfratto commented:
> Also, now the collector/exporter/otlpexporter and collector/exporter/otlphttpexporter packages got their own go.mod file and are now treated as separate module roots, so two extra replace directives were required for us to continue using our fork.

Were we importing anything internal from those before? If not, we may be able to use the v0.63.1 versioned modules for those.

rfratto commented on component/otelcol/exporter/prometheus/internal/convert/convert_test.go:
```diff
@@ -318,9 +318,10 @@ func TestConverter(t *testing.T) {
 		},
 	}
 
+	decoder := &pmetric.JSONUnmarshaler{}
```
Does this keep any state? Is it safe to share across all tests?

tpaschalis replied:
No, the `JSONUnmarshaler` is an empty struct, at least for now. We recreate it just in case this changes in the future, but for now it's fine.

rfratto commented:
Don't forget to update the CHANGELOG before merging 👍

tpaschalis commented:
cc @rfratto @mattdurham @mapno There is one slighly bigger change when updating this dependency, wanted to spell it out in case we encounter it later on.

The upgrade breaks a lot of our tests in `pkg/traces/config_test.go`; if a config had no processors and no extensions, they would be a nil map, and absent from the final config and whereas now, they would appear in the final config as an an empty map instead.
```
|| 	            	Diff:
|| 	            	--- Expected
|| 	            	+++ Actual
|| 	            	- Processors: (map[config.ComponentID]config.Processor) <nil>,
|| 	            	- Extensions: (map[config.ComponentID]config.Extension) <nil>,
|| 	            	+ Processors: (map[config.ComponentID]config.Processor) {
|| 	            	+ },
|| 	            	+ Extensions: (map[config.ComponentID]config.Extension) {
|| 	            	+ },
```

I _think_ the reason is [this](https://github.com/open-telemetry/opentelemetry-collector/commit/9a4333959bbbbb8722b67a92713a0c37bf3a1e13) commit moved around some of the decoding logic and added an extra processing [step](https://github.com/open-telemetry/opentelemetry-collector/commit/9a4333959bbbbb8722b67a92713a0c37bf3a1e13#diff-cfb994fde71919bae57bfbdf7aa927d639793a0d187a59d4c9803af49c74c376R39-R41) which ultimately calls to a [expandNilStructPointersHookFunc](https://github.com/open-telemetry/opentelemetry-collector/blob/bab8fdab1bda5502911294abe47674fbc78d3a11/confmap/confmap.go#L192-L207) that expands values that map to nil pointers.

I've changed our tests to add the new fields, I don't think there's a good reason to not follow upstream and keep the old behavior around.
